source/extensions/formatter/cel/cel.cc

```diff
@@ -0,0 +1,89 @@
+#include "source/extensions/formatter/cel/cel.h"
+
+#include "source/common/config/metadata.h"
+#include "source/common/formatter/substitution_formatter.h"
+#include "source/common/http/utility.h"
+#include "source/common/protobuf/utility.h"
+
+#if defined(USE_CEL_PARSER)
+#include "parser/parser.h"
+#endif
+
+namespace Envoy {
+namespace Extensions {
+namespace Formatter {
+
+namespace {
+
+void truncate(std::string& str, absl::optional<size_t> max_length) {
+  if (!max_length) {
+    return;
+  }
+
+  str = str.substr(0, max_length.value());
+}
+
+} // namespace
+
+namespace Expr = Filters::Common::Expr;
+
+CELFormatter::CELFormatter(Expr::Builder& builder,
+                           const google::api::expr::v1alpha1::Expr& input_expr,
+                           absl::optional<size_t>& max_length)
+    : parsed_expr_(input_expr), max_length_(max_length) {
+  compiled_expr_ = Expr::createExpression(builder, parsed_expr_);
+}
+
+absl::optional<std::string> CELFormatter::format(const Http::RequestHeaderMap& request_headers,
+                                                 const Http::ResponseHeaderMap& response_headers,
+                                                 const Http::ResponseTrailerMap& response_trailers,
+                                                 const StreamInfo::StreamInfo& stream_info,
+                                                 absl::string_view,
+                                                 AccessLog::AccessLogType) const {
+  Protobuf::Arena arena;
+  auto eval_status = Expr::evaluate(*compiled_expr_, arena, stream_info, &request_headers,
+                                    &response_headers, &response_trailers);
+  if (!eval_status.has_value() || eval_status.value().IsError()) {
+    return absl::nullopt;
+  }
+  auto result = Expr::print(eval_status.value());
+  truncate(result, max_length_);
```

Comment:
Sorry, this is probably my fault for not explaining it clearly. Modifying by reference is not desirable - the usual style is to be explicit about mutation via pointer.

Can we change it simply to 
```c++
if  (max_length_) {
  return result.substr(0, *max_length_);
}
```

Reply:
done, thanks for your patient review.